File: JavaScript/6kyu/_testing.js
import { strict as assert } from "assert";
import { likes } from "./WhoLikesIt.js";
import { toCamelCase } from "./ConvertStringToCamelCase.js";

describe("6kyu Tests", () => {
  it("Who Likes It?", function () {
    assert.strictEqual(likes([]), "no one likes this");
    assert.strictEqual(likes(["Peter"]), "Peter likes this");
    assert.strictEqual(likes(["Jacob", "Alex"]), "Jacob and Alex like this");
    assert.strictEqual(
      likes(["Max", "John", "Mark"]),
      "Max, John and Mark like this"
    );
    assert.strictEqual(
      likes(["Alex", "Jacob", "Mark", "Max"]),
      "Alex, Jacob and 2 others like this"
    );
  });
  it("Convert string to camel case", () => {
    assert.strictEqual(
      toCamelCase(""),
      "",
      "An empty string was provided but not returned"
    );
    assert.strictEqual(
      toCamelCase("the_stealth_warrior"),
      "theStealthWarrior",
      "toCamelCase('the_stealth_warrior') did not return correct value"
    );
    assert.strictEqual(
      toCamelCase("The-Stealth-Warrior"),
      "TheStealthWarrior",
      "toCamelCase('The-Stealth-Warrior') did not return correct value"
    );
    assert.strictEqual(
      toCamelCase("A-B-C"),
      "ABC",
      "toCamelCase('A-B-C') did not return correct value"
    );
  });
});

File: Python/6kyu/_testing.py
import unittest
from WhoLikesIt import likes


class Testing_6kyu(unittest.TestCase):
    def test_who_likes_it(self):
        self.assertEqual(likes([]), 'no one likes this')
        self.assertEqual(likes(['Peter']), 'Peter likes this')
        self.assertEqual(likes(['Jacob', 'Alex']),
                           'Jacob and Alex like this')
        self.assertEqual(likes(['Max', 'John', 'Mark']),
                           'Max, John and Mark like this')
        self.assertEqual(
            likes(['Alex', 'Jacob', 'Mark', 'Max']), 'Alex, Jacob and 2 others like this')


if __name__ == "__main__":
    unittest.main()
